blazor/dropdown-list/data-binding.md

```diff
@@ -7,68 +7,377 @@ control: DropDown List
+
+{% include_relative code-snippet/data-binding/web-api-adaptor.razor %}
+
+{% endhighlight %}
+
+![Blazor DropdownList with Web API Adaptor](./images/data-binding/blazor_dropdown_web-api-adaptor.png)
 
-    List<Country> Countries = new List<Country>
+### Custom adaptor
+
+The [SfDataManager](https://help.syncfusion.com/cr/aspnetcore-blazor/Syncfusion.Blazor.Data.SfDataManager.html) has custom adaptor support which allows you to perform manual operations on the data. This can be utilized for implementing custom data binding and editing operations in the DropDownList component.
+
+For implementing custom data binding in DropDownList, the `DataAdaptor` class is used. This abstract class acts as a base class for the custom adaptor.
+
+The `DataAdaptor` abstract class has both synchronous and asynchronous method signatures which can be overridden in the custom adaptor. Following are the method signatures present in this class,
+
+```csharp
+public abstract class DataAdaptor
 {
-        new Country() { Name = "Australia", Code = "AU" },
-        new Country() { Name = "Bermuda", Code = "BM" },
-        new Country() { Name = "Canada", Code = "CA" },
-        new Country() { Name = "Cameroon", Code = "CM" },
-    };
+    /// <summary>
+    /// Performs data Read operation synchronously.
+    /// </summary>
+    public virtual object Read(DataManagerRequest dataManagerRequest, string key = null)
+
+    /// <summary>
+    /// Performs data Read operation asynchronously.
+    /// </summary>
+    public virtual Task<object> ReadAsync(DataManagerRequest dataManagerRequest, string key = null)
 }
 ```
 
-## Index Value Binding
+The custom data binding can be performed in the DropDownList component by providing the custom adaptor class and overriding the Read or ReadAsync method of the DataAdaptor abstract class.
+
+The following sample code demonstrates implementing custom data binding using custom adaptor,
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/data-binding/custom-adaptor.razor %}
+
+{% endhighlight %}
+
+### Offline mode
+
+To avoid post back for every action, set the DropDownList to load all data on initialization and make the actions process in client-side. To enable this behaviour, use the `Offline` property of `DataManager`.
+
+The following example for remote data binding and enabled offline mode.
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/data-binding/offline-mode.razor %}
+
+{% endhighlight %}
+
+![Blazor DropdownList with Offline mode](./images/data-binding/blazor_dropdown_offline-mode.png)
+
+### Entity framework
 
-Index value binding can be achieved by using `bind-Index` attribute and it supports int and int nullable types. By using this attribute you can bind the values respective to its index.
+You need to follow the below steps to consume data from the [Entity Framework](https://blazor.syncfusion.com/documentation/common/data-binding/bind-entity-framework) in the DropDownList component.
```

Comment:
Follow these steps to consume data from the [Entity Framework](https://blazor.syncfusion.com/documentation/common/data-binding/bind-entity-framework) in the DropDownList component.